### ISSUE TITLE
Less prerequisites for instrumentation

### ISSUE THREAD

**Issue by ffMathy** [lucaslorentz/minicover#137]
Continuing on the premise and use-case described in #136, I noticed that the following options seem required (from the documentation):

- `sources`
- `tests`

But why are these required? What are they used for? I don't currently need to specify it when using AltCover, and I'd like to move away from it and over to Minicover instead.

Is it possible to infer these values? AltCover must be doing the same somehow.

So to summarize all my questions:
1. Why are these required?
2. What are they used for?
3. Can we infer them instead?

**Comment by ffMathy** [comment]
The dilemma is that the users of my Pruner CLI shouldn't need to specify these things if it can be avoided. Today, it just needs a directory pointing to your project or solution, and it'll run `dotnet test` from that, along with AltCover instrumentation etc.

It'd be great if this could somehow do the same.

**Comment by ffMathy** [comment]
@lucaslorentz investigating further, it seems AltCover just checks if the source file for a given assembly exists on disk, and then assumes instrumentation is needed.

That would be a very cool default if none of the other options were specific, or at least make it possible if using `Minicover.Core`. Once that's solved, I can start using the project!

**Comment by lucaslorentz** [comment]
Makes sense, I will make MiniCover.Core more flexible regarding that

**Comment by lucaslorentz** [comment]
Probably later I will improve the tool experience as well.

**Comment by ffMathy** [comment]
Great! Any ETA on this? Can't wait! 😍

**Comment by lucaslorentz** [comment]
As a temporary workaround, you can create a class extending InstrumentationContext, override IsSourceFile and do a File.Exists inside it.
Something similar for IsTestFile, if you have any way of knowing if the file is a test file or not.

**Comment by ffMathy** [comment]
But will that be enough?

Aren't those arguments used implicitly somewhere else too?

**Comment by lucaslorentz** [comment]
No, they're not.
Can you please review? https://github.com/lucaslorentz/minicover/pull/138

**Comment by ffMathy** [comment]
Wow, already! Thank you so much :pray:

I left a comment!

**Comment by ffMathy** [comment]
Also, do you want to connect perhaps? Maybe on Discord or Skype?

I think we may have a very similar coding style, and it looks like we both like Open Source :pray:

**Comment by lucaslorentz** [comment]
Sounds good, I'm completely new to discord. Created this server for minicover: https://discord.gg/WqBBjnZcCe
I will learn as we go
